src/DisplayPlug.vala

```diff
@@ -182,6 +210,10 @@ public class Display.Plug : Switchboard.Plug {
                     critical (e.message);
                 }
             });
+
+            rotation_lock_switch.notify["active"].connect (() => {
```

Comment:
What you wrote here is actually an already available functionality in `GLib.Settings`.

What you can do here is to use the `bind` method of the Settings to synchronize both values so they change based on each other. You can do so, by calling `bind` like this and removing the entire signal notifier:
`rotation_lock_setting.bind ("orientation-lock", rotation_lock_switch, "active", SettingsBindFlags.DEFAULT);`